query-node/joystream-query-node/schema.graphql

```diff
@@ -1,21 +1,489 @@
+	defaultEntityCreationVoucherUpperBound: BigInt!
+
+	#### ClassPermissions ####
+
+	"For this permission, the individual member is allowed to create the entity and become controller"
+	anyMemberCanCreate: Boolean!
+
+	"Whether to prevent everyone from creating an entity"
+	entityCreationBlocked: Boolean!
+
+	"Whether to prevent everyone from updating entity properties"
+	allEntityPropertyValuesLocked: Boolean!
+
+	"Current class maintainer curator groups"
+	maintainers: [CuratorGroup!]
+
+	#### ClassPermissions ####
+
+	# Commenting out this line because the properties are accessible from the schemas
+	# query { classes { schemas {properties { ... } } } }
+	# "All properties that have been used on this class across different class schemas"
+	# properties: [Property!]
+
+	"All schemas that are available for this class, think v0.0 Person, v.1.0 Person, etc."
+	schemas: [Schema!] @derivedFrom(field: "class")
+
+	"All vouchers for this class"
+	vouchers: [EntityCreationVoucher!] @derivedFrom(field: "class")
+
+	"Entities associated with the clas"
+	entities: [Entity!] @derivedFrom(field: "class")
+}
+
+enum PropertyType {
+	BOOL
+	UINT16
+	UINT32
+	UINT64
+	INT16
+	INT32
+	INT64
+	TEXT
+	HASH
+	REFERENCE
+	VECBOOL
+	VECUINT16
+	VECUINT32
+	VECUINT64
+	VECINT16
+	VECINT32
+	VECINT64
+	VECTEXT
+	VECREFERENCE
+}
+
+"`Property` representation, related to a given `Class`"
+type Property @entity {
+	"Runtime identifier for property in class"
+	propertyId: BigInt! @unique
+
+	"Type of `Property`"
+	propertyType: PropertyType!
+
+	"If property value can be skipped, when adding entity schema support"
+	required: Boolean!
+
+	"Used to enforce uniqeness of a property across all entities that have this property"
+	unique: Boolean!
+
+	"Name of property"
+	name: String!
+
+	"Description of property"
+	description: String!
+
+	"Property is locked from maintainer"
+	isLockedFromMaintainer: Boolean!
+
+	"Property is locked from controller"
+	isLockedFromController: Boolean!
+
+	"Upper bound for vector length when property is a vector"
+	maximumLength: BigInt
+
+	"Optional field, it must reference a class then PropertyType is Reference"
+	referencedClass: Class
+
+	"Optional field, Only will be set when PropertyType is Reference"
+	sameControllerStatus: Boolean
+
+	"Schemas which involve this property"
+	schemas: [Schema!] @derivedFrom(field: "properties")
+}
+
+"Represents a standard for what information must be associated with any entity of a given class"
+type Schema @entity {
+	"Runtime identifier of schema in class"
+	schemaId: BigInt! @unique
+
+	"Class to which this schema corresponds"
+	class: Class!
+
+	"Properties of class that are in this schema"
+	properties: [Property!]
+
+	"If schema can be added to an entity"
+	isActive: Boolean!
+
+	"Virtual field, supported entities"
+	entities: [Entity!] @derivedFrom(field: "supportedSchemas")
+}
+
+"Working group participant: working-group/src/types.rs"
+type Worker @entity {
```

Comment:
This should probably be an interface, as there are many kinds of workers, not just curators (even though we will only include curators in Babylon). The `Curator` entity can implement this interface. This would also require dropping `curator` field.

Reply:
Done in 9540c34